packages/frontend/src/components/ChartConfigPanel/FieldLayoutOptions.tsx

```diff
@@ -1,62 +1,143 @@
-import { Button, ButtonGroup, Icon } from '@blueprintjs/core';
+import { Button } from '@blueprintjs/core';
+import { Field, getItemId, isField, TableCalculation } from 'common';
+import React, { FC, useEffect, useState } from 'react';
+import FieldAutoComplete from '../common/Filters/FieldAutoComplete';
+import { useVisualizationContext } from '../LightdashVisualization/VisualizationProvider';
 import {
-    Field,
-    getItemColor,
-    getItemIcon,
-    isField,
-    TableCalculation,
-} from 'common';
-import React, { FC } from 'react';
-import { GridFieldLabel } from './ChartConfigPanel.styles';
+    AxisFieldDropdown,
+    AxisGroup,
+    AxisTitle,
+    DeleteFieldButton,
+} from './ChartConfigPanel.styles';
 
 type Props = {
-    item: Field | TableCalculation;
-    isXActive?: boolean;
-    isYActive?: boolean;
-    isGroupActive?: boolean;
-    onXClick: (isActive: boolean) => void;
-    onYClick: (isActive: boolean) => void;
-    onGroupClick: (isActive: boolean) => void;
+    items: (Field | TableCalculation)[];
 };
 
-const FieldLayoutOptions: FC<Props> = ({
-    item,
-    isXActive,
-    isGroupActive,
-    onGroupClick,
-    onXClick,
-    onYClick,
-    isYActive,
-}) => {
+const FieldLayoutOptions: FC<Props> = ({ items }) => {
+    const {
+        cartesianConfig: {
+            dirtyLayout,
+            setXField,
+            addSingleSeries,
+            removeSingleSeries,
+            updateYField,
+        },
+        pivotDimensions,
+        setPivotDimensions,
+    } = useVisualizationContext();
+    const pivotDimension = pivotDimensions?.[0];
+    const [yFieldsKeys, setYFieldKeys] = useState(dirtyLayout?.yField || []);
+
+    useEffect(() => {
+        if (dirtyLayout) {
+            setYFieldKeys(dirtyLayout?.yField || []);
+        }
+    }, [dirtyLayout?.yField]);
+
+    // X axis logic
+    const xAxisField = items.find(
+        (item) => getItemId(item) === dirtyLayout?.xField,
+    );
+
+    const onXClick = (itemId: any) => {
+        const isActive = xAxisField && getItemId(xAxisField) === itemId;
```

Comment:
I don't think we need the unselect logic here anymore since there should always be a x field.